packages/-ember-data/tests/integration/identifiers/lid-reflection-test.ts

```diff
@@ -220,4 +220,76 @@ module('Integration | Identifiers - lid reflection', function(hooks) {
     assert.deepEqual(cake.hasMany('ingredients').ids(), ['2']);
     assert.equal(cake.ingredients.objectAt(0).name, 'Cheese');
   });
+
+  test('belongsTo() has correct state after .save() on a newly created record with sideposted child record when lid is provided in the response payload', async function(assert) {
+    class Topping extends Model {
+      @attr name;
+    }
+
+    class Cake extends Model {
+      @attr name;
+      @belongsTo('topping', { inverse: null, async: false }) topping;
+    }
+
+    this.owner.register('model:topping', Topping);
+    this.owner.register('model:cake', Cake);
+
+    class TestSerializer extends Serializer {
+      normalizeResponse(_, __, payload) {
+        return payload;
+      }
+    }
+    class TestAdapter extends Adapter {
+      createRecord(store, ModelClass, snapshot) {
+        return resolve({
+          data: {
+            type: 'cake',
+            id: '1',
+            attributes: {
+              name: 'Cheesecake',
+            },
+            relationships: {
+              topping: {
+                data: {
+                  type: 'topping',
+                  id: '2',
+                  lid: cheeseIdentifier.lid,
+                },
+              },
+            },
+          },
+          included: [
+            {
+              type: 'topping',
+              id: '2',
+              lid: cheeseIdentifier.lid,
+              attributes: {
+                name: 'Cheese',
+              },
+              relationships: {
+                cake: {
+                  data: {
+                    type: 'cake',
+                    id: '1',
+                  },
+                },
+              },
+            },
+          ],
+        });
+      }
+    }
+    this.owner.register('serializer:application', TestSerializer);
+    this.owner.register('adapter:application', TestAdapter);
+
+    const cheese = store.createRecord('topping', { name: 'Cheese' });
+    const cake = store.createRecord('cake', { name: 'Cheesecake', topping: cheese });
+
+    const cheeseIdentifier = recordIdentifierFor(cheese);
```

Comment:
depending on these costs getting hoisted is an anti pattern we should avoid, move these declarations above their first usage (in TestAdapter)

Reply:
(alternatively declare `cheeseIdentifier` as a `let` above and then assign it here. Either is fine but hoisting an assignment is bad.

Reply:
Absolutely agree, done.

Reply:
@azhiv I think you forgot to push the changes up

Reply:
@runspired In fact I committed and pushed to another branch, accidentally. The changes are now on master.

Reply:
@runspired Both adapter and serializer need to be registered before calling `createRecord`, otherwise those will be resolved to default json-api implementations, which will then be stored in ED's cash. This modification breaks the tests, I reverted the change. The tests pass now.

Reply:
@runspired this required an additional fix. The order of method calls is crucial.